applicationset/utils/createOrUpdate.go

```diff
@@ -88,7 +88,7 @@ func CreateOrUpdate(ctx context.Context, c client.Client, obj client.Object, f c
 // mutate wraps a MutateFn and applies validation to its result
 func mutate(f controllerutil.MutateFn, key client.ObjectKey, obj client.Object) error {
 	if err := f(); err != nil {
-		return err
+		return fmt.Errorf("error mutate wraps a MutateFn: %w", err)
```

Comment:
```suggestion
		return err
```

I think in this case it's fine to just return the error.